File: backend/internal/putg/repository/postgres/pg_models/material.go
package pg_models

type Material struct {
	Id         string `db:"id"`
	MaterialId string `db:"material_id"`
	Type       string `db:"type"`
	IsDefault  bool   `db:"is_default"`
	Code       string `db:"code"`
	IsStandard bool   `db:"is_standard"`
	BaseCode   string `db:"base_code"`
	Title      string `db:"title"`
}

File: frontend/src/features/gaskets/modules/snp/types/material.ts
import { IMaterial } from '@/features/gaskets/types/material'

export interface IMaterialData {
	filler: IFiller
	frame?: IMaterial
	innerRing?: IMaterial
	outerRing?: IMaterial
	openFiller: boolean
	openIr: boolean
	openFr: boolean
	openOr: boolean
}

export interface IMaterialDataDTO {
	fillerId: string
	frameId: string
	innerRingId: string
	outerRingId: string
}

export interface IFiller {
	id: string
	temperature: string
	baseCode: string
	code: string
	title: string
	description: string
	designation: string
	disabledTypes?: string[]
}

export type TypeMaterial = 'outerRing' | 'frame' | 'innerRing'
export type OpenMaterial = 'filler' | TypeMaterial

export interface ISnpMaterials {
	frame: IMaterial[]
	innerRing: IMaterial[]
	outerRing: IMaterial[]
	frameDefaultIndex?: number
	innerRingDefaultIndex?: number
	outerRingDefaultIndex?: number
}
